ereolen_support/ereolen_support.install

```diff
@@ -0,0 +1,197 @@
+    $vocabulary->machine_name = "ereolen_support_device";
+    $vocabulary->description = t('List of devices used in support form');
+    $vocabulary->module = 'taxonomy';
+
+    taxonomy_vocabulary_save($vocabulary);
+    $vid = $vocabulary->vid;
+    _ereolen_support_create_terms($device_terms, $vid);
+  }
+}
+
+/**
+ * Create terms programmatically.
+ *
+ * @param array $termnames
+ *   A list of term names to add to a vocabulary.
+ * @param $vid
+ *   The vocabulary id.
+ */
+function _ereolen_support_create_terms($termnames, $vid) {
+  foreach ($termnames as $termname) {
+    // Create terms.
+    $newterm = new stdClass();
+    $newterm->name = $termname;
+    $newterm->vid = $vid;
+    $newterm->parent = 0;
+    taxonomy_term_save($newterm);
+  }
+}
+
+/**
+ * Implements hook_schema().
+ */
+function ereolen_support_schema() {
+  $schema['ereolen_support_errors'] = [
+    'description' => 'Table for errors during support form submit',
+    'fields' => [
+      'id' => [
+        'description' => 'The primary identifier for an entry',
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'timestamp' => [
+        'description' => 'The time of the error',
+        'type' => 'varchar',
+        'length' => 32,
+        'not null' => TRUE,
+        'default' => '',
+      ],
+      'request' => [
+        'description' => 'The request sent to Jira',
+        'type' => 'varchar',
+        'length' => 2048,
+        'not null' => TRUE,
+        'default' => '',
+      ],
+      'data' => [
+        'type' => 'blob',
+        'not null' => TRUE,
+        'size' => 'big',
+        'description' => 'Serialized array of data array that resulted in error',
+      ],
+    ],
+    'primary key' => [
+      'id',
+    ],
+  ];
+  return $schema;
+}
+
+/**
+ * Implements hook_uninstall().
+ */
+function ereolen_support_uninstall() {
+  $custom_variables = array(
+    'ereolen_support_library_description',
+    'ereolen_support_library_required',
+    'ereolen_support_name_description',
+    'ereolen_support_name_required',
+    'ereolen_support_email_description',
+    'ereolen_support_email_required',
+    'ereolen_support_id_description',
+    'ereolen_support_id_required',
+    'ereolen_support_product_description',
+    'ereolen_support_product_description_required',
+    'ereolen_support_problem_description',
+    'ereolen_support_problem_required',
+    'ereolen_support_description_description',
+    'ereolen_support_description_required',
+    'ereolen_support_date_description',
+    'ereolen_support_date_required',
+    'ereolen_support_book_title_description',
+    'ereolen_support_book_title_required',
+    'ereolen_support_device_description',
+    'ereolen_support_device_required',
+    'ereolen_support_model_description',
+    'ereolen_support_model_required',
+    'ereolen_support_operating_system_description',
+    'ereolen_support_model_operating_system',
+    'jira_domain',
```

Comment:
Nice-to-have: All variables should be prefixed with the module name :-) I don't think we will have other modules on this site that would conflict with this... so nice-to-have.

Reply:
Message received